assets-src/js/main/nav-single-level.js

```diff
@@ -0,0 +1,83 @@
+import './_object.assign.polyfill';
+
+/**
+ * Object for creating single-level navigation menus
+ * Manages button for toggling navigation on mobile
+ * Uses event delegation to handle events for improved performance
+ *
+ * @param {Element} menu - the top level navigation <ul>
+ * @param {Object} options - configuration options for the navigation
+ * @param {number} [options.breakpoint=1024] - pixel value at which the button for toggling the mobile navigation is hidden. Is converted to em.
+ * @param {string} [options.mobileIcon] - SVG icon used for the button to show/hide the navigation on mobile.
+ */
+
+const navSingleLevel = function(menu, options) {
+	let	container = menu.parentElement;
+	let mobileToggle = document.querySelector('[data-trigger="mobile-nav"]');
+
+	// Default settings
+	let defaults = {
+		breakpoint: 1024,
+		mobileIcon: '<svg xmlns="http://www.w3.org/2000/svg" height="24" width="24" class="icon icon--24" focusable="false" aria-hidden="true" fill="currentColor">' +
+			'<path class="open" d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"/>' +
+			'<path class="close" d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"/>' +
+			'</svg>'
+	};
+
+	// Merge user options into defaults
+	let settings = Object.assign({}, defaults, options);
+
+	this.init = function() {
+		mobileToggleSetup();
+		document.addEventListener('click', clickHandler);
+		document.addEventListener('keyup', closeOnEscKey);
+	}
+
+	function clickHandler(event) {
+		if (event.target.matches('[data-trigger="mobile-nav"]')) {
```

Comment:
Hi Nicki,

I'm using this on the Welsh Boundary Commission website, and noted something about how this code works.

The button includes a hidden span element, the docs state you can just remove the class that hides it to use it as a label, e.g.

```
<button class="button button--ghost" data-trigger="mobile-nav" style="display: none;">
    <span>Menu</span>
</button>
```

The problem is that doing this doesn't work if you try to click/select the "Menu" text in browser, this is because this event handler only checks for matches on the precise target of the click event (so if you click the span element, it will look for `[data-trigger="mobile-nav"]` on the span, and not the button, so the check will fail and the nav won't open).

I can see three ways to fix this:

1. Remove the span element, and have the text as just button text. This works in my instance, but there might be cases where we want a span element (or another tag maybe) within the `<button>` element for some reason

2. We can change the JavaScript code to use `closest` instead of `matches`. This will ensure any child elements that are beneath the button will also trigger this code when they are selected:

```
function clickHandler(event) {
	var menuButton = event.target.closest('[data-trigger="mobile-nav"]');

	if (menuButton) {
		if (menuButton.matches('[aria-expanded="true"]')) {
			menuButton.setAttribute('aria-expanded', 'false');
		} else {
			menuButton.setAttribute('aria-expanded', 'true');
		}
	}
}
```
(https://developer.mozilla.org/en-US/docs/Web/API/Element/closest)

3. Child elements of the button could be set to use the `pointer-events: none` CSS like the child SVG. This would be a simpler solution but feels a bit more fragile to me as it would rely on the developer setting this up? I don't think there are any accessibility implications to doing it this way, but I could be wrong.

Reply:
Thanks a lot for this really helpful feedback @sean-dunwoody I'm going to dig into this further before we get anything merged

Reply:
Hi @sean-dunwoody - I have decided to go with option 3, because all it requires is applying the existing `pointer-events: none` style to any child element rather than just to SVGs. I don't think it's fragile, as it's effectively already taken care of for developers.